src/ampycloud/layer.py

```diff
@@ -221,9 +221,9 @@ def ncomp_from_gmm(vals: np.ndarray,
     models = {}
 
     # Run the Gaussian Mixture fit for all cases ... should we do anything more fancy here ?
-    with utils.tmp_seed(random_seed):
-        for n_val in ncomp:
-            models[n_val] = GaussianMixture(n_val, covariance_type='spherical').fit(vals)
+    for n_val in ncomp:
+        models[n_val] = GaussianMixture(n_val, covariance_type='spherical',
+                                        random_state=random_seed).fit(vals)
```

Comment:
Well, the layering step is definitely important. It is also robust. And in fact I think we did set the random seed in a "ok-ish" way, because there is a dedicated test for it (`def test_unstable_layers()` in `test_layers.py`). 

However, I think the way it was set before was not the most ideal, i.e. it would not carry over if one interrupts with a `pdb.set_trace()`. The new approach (feeding the random seed directly to `GaussianMixture()`) does work in that case.